compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUPromoteMatmulToFitMMA.cpp

```diff
@@ -107,7 +104,40 @@ class LLVMGPUPromoteMatmulToFitMMAPass final
 
     IRRewriter rewriter(ctx);
     for (auto op : candidates) {
-      padWithZeroValue(rewriter, op, targetIterType, nofold);
+      FailureOr<linalg::ContractionDimensions> dims =
+          linalg::inferContractionDims(op);
+      assert(succeeded(dims) && "expected contraction op interface");
+
+      // Populate padding dimensions.
+      SmallVector<int64_t> paddingDimensions;
+      for (auto [idx, iter] : llvm::enumerate(op.getIteratorTypesArray())) {
+        if (iter == targetIterType) {
+          paddingDimensions.push_back(idx);
+        }
+      }
+
+      // Populate tile sizes. All tile sizes are one, except mma tile sizes.
```

Comment:
Can you comment on what the desired padding is here? IE it's clear that only mma tile sizes are non-one, but  I think it would be more informative to say what the intention/end-state is.

If I'm reading this right, you want to pad to the multiple of the workgroup-level tile?

Reply:
Fixed. It should have been padded to multiples of workgroup/reduction.